exampleCustomVideoCapturer/src/main/java/com/twilio/video/examples/customcapturer/ViewCapturer.kt

```diff
@@ -0,0 +1,124 @@
+package com.twilio.video.examples.customcapturer
+
+import android.content.Context
+import android.graphics.Bitmap
+import android.graphics.Canvas
+import android.os.Handler
+import android.os.Looper
+import android.os.SystemClock
+import android.view.View
+import com.twilio.video.Rgba8888Buffer
+import com.twilio.video.VideoCapturer
+import com.twilio.video.VideoDimensions
+import com.twilio.video.VideoFormat
+import tvi.webrtc.CapturerObserver
+import tvi.webrtc.SurfaceTextureHelper
+import tvi.webrtc.VideoFrame
+import java.nio.ByteBuffer
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.atomic.AtomicBoolean
+
+/**
+ * ViewCapturer demonstrates how to implement a custom [VideoCapturer]. This class
+ * captures the contents of a provided view and signals the [tvi.webrtc.CapturerObserver] when
+ * the frame is available.
+ */
+class ViewCapturer(private val view: View) : VideoCapturer {
+    private val handler = Handler(Looper.getMainLooper())
+    private var capturerObserver: CapturerObserver? = null
+    private val started =
+        AtomicBoolean(false)
+
+    private val viewCapturer = {
+        val dropFrame = view.width == 0 || view.height == 0
+
+        // Only capture the view if the dimensions have been established
+        if (!dropFrame) {
+            // Draw view into bitmap backed canvas
+            val measuredWidth = View.MeasureSpec.makeMeasureSpec(
+                view.width,
+                View.MeasureSpec.EXACTLY
+            )
+            val measuredHeight = View.MeasureSpec.makeMeasureSpec(
+                view.height,
+                View.MeasureSpec.EXACTLY
+            )
+            view.measure(measuredWidth, measuredHeight)
+            view.layout(0, 0, view.measuredWidth, view.measuredHeight)
+            val viewBitmap = Bitmap.createBitmap(
+                view.width, view.height,
+                Bitmap.Config.ARGB_8888
+            )
+            val viewCanvas = Canvas(viewBitmap)
+            view.draw(viewCanvas)
+
+            // Extract the frame from the bitmap
+            val bytes = viewBitmap.byteCount
+            val buffer = ByteBuffer.allocate(bytes)
+            viewBitmap.copyPixelsToBuffer(buffer)
+
+            // Create video frame
+            val captureTimeNs =
+                TimeUnit.MILLISECONDS.toNanos(SystemClock.elapsedRealtime())
+            val videoBuffer: VideoFrame.Buffer =
+                Rgba8888Buffer(buffer, view.width, view.height)
+            val videoFrame = VideoFrame(videoBuffer, 0, captureTimeNs)
+
+            // Notify the observer
+            if (started.get()) {
+                capturerObserver!!.onFrameCaptured(videoFrame)
```

Comment:
Maybe use a null safe operator and throw a more detail exception if the reference is null?